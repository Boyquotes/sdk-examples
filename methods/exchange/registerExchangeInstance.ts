import { ExecuteMethod } from "../../executeMethod";

export const executeMethod: ExecuteMethod = async ({ exchangeInstanceId }) => {
  return exchangeInstanceId;
};
