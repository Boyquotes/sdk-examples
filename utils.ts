import CedeSDK from "@cedelabs/demo-sdk";
import { exchanges } from "./exchanges";

const SDK_MODE = "MOCK";
const SELECTED_EXCHANGE = "bybit";

const clientId = "sdk-***"; // update with your clientId if using with "REAL" SDK_MODE

export const cedeSDK = new CedeSDK(SDK_MODE ?? "MOCK");

export const initExchange = async (specificExchange = "coinbase") => {
  const selectedExchange = SELECTED_EXCHANGE ?? specificExchange;

  function assertIsExchangeKey(key: string): key is keyof typeof exchanges {
    return key in exchanges;
  }

  if (!assertIsExchangeKey(selectedExchange)) {
    throw new Error(`Exchange ${selectedExchange} not found`);
  }

  const exchange = exchanges[selectedExchange];
  const registeredExchange = await cedeSDK.api.registerExchangeInstance({
    exchangeId: SELECTED_EXCHANGE ?? specificExchange,
    ...exchange,
  });

  return registeredExchange.exchangeInstanceId;
};
