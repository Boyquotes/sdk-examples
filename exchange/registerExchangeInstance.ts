import { exchanges } from "../exchanges";
import { cedeSDK } from "../utils";

export const executeMethod = async () => {
  const { registerExchangeInstance } = cedeSDK.api;

  const exchange = exchanges[process.env.SELECTED_EXCHANGE ?? ""];
  const registeredExchange = await registerExchangeInstance({
    exchangeId: process.env.SELECTED_EXCHANGE ?? "",
    ...exchange,
  });

  return registeredExchange;
};
