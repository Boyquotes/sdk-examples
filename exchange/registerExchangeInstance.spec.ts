import { initExchange } from "../utils";
import { executeMethod } from "./registerExchangeInstance";

describe("registerExchangeInstance", () => {
  beforeAll(async () => {
    await initExchange();
  });

  it("it should return exchange instance id", async () => {
    const result = await executeMethod();

    expect(result).toMatchInlineSnapshot(`
      {
        "exchangeInstanceId": "133271-623848-295882-498157-147268",
      }
    `);
  });
});
