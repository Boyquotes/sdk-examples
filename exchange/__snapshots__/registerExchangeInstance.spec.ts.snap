// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`registerExchangeInstance it should return exchange instance id 1`] = `
{
  "exchangeInstanceId": Any<String>,
}
`;
